### PR TITLE
Handle the implicit/default label in a switch statement in the nullable walker.

### LINKED ISSUE #40629: Pattern-matching null check in switch does not update nullability after the switch
**Version Used**: 16.4.2

```cs
#nullable enable

class C
{
    void M(object? p)
    {
        switch (p)
        {
            case null:
                return;
        }

        // CS8602 Dereference of a possibly null reference.
        p.ToString();
    }
}
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #40629

### CODE REVIEW

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a couple of reviews of this small change?

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs [hunk 1]
```diff
@@ -2071,5 +2071,28 @@ static void M1(C c, C c2)
...
+        [WorkItem(40629, "https://github.com/dotnet/roslyn/issues/40629")]
+        public void NullTestInSwitch_01()
+        {
+            CSharpCompilation c = CreateNullableCompilation(@"
+#nullable enable
+class C
+{
+    void M(object? p)
+    {
+        switch (p)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs]: Is it expected that this change would also fix a case like the following?

```cs
switch (p == null)
{
    case true:
        return;
}

p.ToString();
```

Comment (gafter) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs]: I would not expect that to work any more than I would expect this to work:
`if ((p == null) == true) return;`

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs]: Makes sense, thanks.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a review from a Sr engineer?

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a review from a Sr engineer?

**Pull request merged**